backend/fms_core/tests/test_models/property_value_test.py

```diff
@@ -0,0 +1,98 @@
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from django.test import TestCase
+from fms_core.models import PropertyValue, PropertyType, Protocol, Process
+
+
+class PropertyValueTest(TestCase):
+    def setUp(self):
+        self.protocol, _ = Protocol.objects.get_or_create(name="ProtocolTest")
+        self.property_type_with_str = PropertyType.objects.create(name="PropertyNameTestStr",
+                                                                   value_type="str",
+                                                                   content_object=self.protocol)
+        self.property_type_with_int = PropertyType.objects.create(name="PropertyNameTestInt",
+                                                                            value_type="int",
+                                                                            content_object=self.protocol)
+        self.process, _ = Process.objects.get_or_create(protocol=self.protocol)
+
+    def test_property_value(self):
+        content_object = self.process
+        value = 10.1
+        property_type_with_float = PropertyType.objects.create(name="PropertyNameTestFloat",
+                                                                            value_type="float",
+                                                                            content_object=self.protocol)
+        pv = PropertyValue.objects.create(value=value,
+                                          property_type=property_type_with_float,
+                                          content_object=content_object)
+
+        self.assertEqual(pv.property_type, property_type_with_float)
+        self.assertEqual(pv.content_object, content_object)
+        self.assertEqual(pv.value, value)
+        self.assertEqual(type(pv.value).__name__, pv.property_type.value_type)
+
+    def test_incorrect_json_value_schema(self):
+        content_object = self.process
+        value = "testtest"
+        with self.assertRaises(ValidationError):
+            try:
+                PropertyValue.objects.create(value=[value, value],
+                                             property_type=self.property_type_with_int,
```

Comment:
property_type do not match the value ...

Reply:
the point here is to have an invalid json schema (such as [value, value]) the property type could be anything that is valid. However i changed it to str, so it is closer to our value, but it is still not accurate and cant be

Reply:
I mentioned this because if you remove the schema issue you'll still have a property_type issue with the value. If you have more than one error in these tests if the target error do not fire this might get masked by the other one (not in this case, but as a general danger of leaving other errors in those test cases.

Reply:
What i mean is that in this case the type array still wont match with any of the property type (int, float, bool,str, etc.)